### DISCUSSION TITLE
this那篇习题答案有误

### DISCUSSION THREAD

**Discussion by xiaowangge666** [javascript-tutorial/zh.javascript.info#1062] [Q&A] [answered]
在this那篇教程上说下方代码是会报错 因为this是undefined 但是实际不会
 function makeUser() {
            return {
                name: "John",
                ref: this
            };
        }

        let user = makeUser();

        alert(user.ref.age);

**Comment by songhn233** [comment, marked as answer]
Hello，本教程已经[说明](https://zh.javascript.info/strict-mode#wo-men-ying-gai-shi-yong-usestrict-ma)：

> 本教程的所有例子都默认采用严格模式，除非特别指定（非常少）。

在严格模式下 global 不会把 this 设置为 globalThis，而是为 undefined

P.S. 同时可以把这个理解为默认行为，因为在现代前端开发绝大部分上下文都是暗含了开启严格模式的。
